app/assets/javascripts/components/PipelineSampleReport.jsx

```diff
@@ -3,358 +3,176 @@ class PipelineSampleReport extends React.Component {
+    this.applyNewFilterThresholds = this.applyNewFilterThresholds.bind(this);
+  }
 
-    this.view_level = ReportFilter.getFilter('view_level') || 'species';
-    this.highest_tax_counts = props.highest_tax_counts;
-    this.defaultSortBy = 'highest_species_nt_zscore';
-    const current_sort = PipelineSampleReport.currentSort();
-    this.state = {
-      sort_query: current_sort.sort_query
-        ? current_sort.sort_query  : `sort_by=${this.defaultSortBy}`
-    };
-    this.applySort = this.applySort.bind(this);
-    this.columnSorting = this.columnSorting.bind(this);
+  refreshPage(overrides) {
+    new_params = Object.assign({}, this.props.report_page_params, overrides);
+    window.location = location.protocol + '//' + location.host + location.pathname + '?' + jQuery.param(new_params);
   }
 
-  static uppCaseFirst(name) {
-    return (name)? name.charAt(0).toUpperCase() + name.slice(1) : name;
+  applyViewLevel(view_level) {
+    this.refreshPage({view_level});
   }
 
-  static currentSort() {
-   const sort_by = ReportFilter.getFilter('sort_by');
-   let current_sort = {};
-   if(sort_by) {
-     current_sort = {
-       sort_query: `sort_by=${sort_by}`
-     }
-   }
-   return current_sort;
+  // applySort needs to be bound at time of use, not in constructor above
+  applySort(sort_by) {
+    this.refreshPage({sort_by});
   }
 
-  applySort(sort_query) {
-    this.setState({ sort_query });
-    const url = PipelineSampleReport.deleteUrlParam(window.location.href, 'sort_by');
-    window.location = (PipelineSampleReport.hasQuery(url))
-      ? `${url}&${sort_query}` : `${url}?${sort_query}`;
+  applyNewFilterThresholds(new_filter_thresholds) {
+    this.refreshPage(new_filter_thresholds);
   }
 
-  static deleteUrlParam(url, parameter) {
-    const queryString = url.split('?');
-    if (queryString.length >= 2) {
-      const prefix = encodeURIComponent(parameter)+'=';
-      const pars = queryString[1].split(/[&;]/g);
-      for (let i = pars.length; i--; i > 0) {
-        if (pars[i].lastIndexOf(prefix, 0) !== -1) {
-          pars.splice(i, 1);
-        }
-      }
-      url = queryString[0] + (pars.length > 0 ? '?' + pars.join('&') : '');
-      return url;
+  render_name(tax_info) {
+    foo = <i>{tax_info.name}</i>;
+    if (tax_info.tax_id > 0) {
+      foo = (
+        <span className="link">
+          <a href={`https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&id=${tax_info.tax_id}`}>
+            {tax_info.name}
+          </a>
+        </span>
+      );
+    }
+    if (tax_info.tax_level == 1) {
+      // indent species rows
+      foo = <span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;{foo}</span>
     } else {
-      return url;
+      // emphasize genus, soften category and species count
+      foo = <span><b>{foo}</b>&nbsp;&nbsp;&nbsp;&nbsp;<span style={{'color':'#A0A0A0'}}><i>({tax_info.species_count}&nbsp;{tax_info.category_name}&nbsp;species)</i></span></span>
```

Comment:
This display feels slightly confusing to me. Would prefer for category name and species-count info to be separated rather than intertwined. Maybe just move category name outside the parenthesis next to the genus name?

Reply:
Let's try that.